### PR TITLE
feat(Date Input): separate date field into three inputs

### PR DESCRIPTION
**Pull request by ingriddorioschulze**

### CODE REVIEW

File: app/services/validation/date.ts [hunk 1]
```diff
@@ -52,3 +122,57 @@ export const createDateSchema = (args?: {
...
+  if (args?.earliest && args?.latest && args.latest() <= args.earliest()) {
+    throw Error(
+      `Latest valid ${args.latest().toDateString()} can't be before earliest valid ${args.earliest().toDateString()}`,
+    );
+  }
+  return z
+    .object({
+      geburtsdatumTag: z.string().min(1, { message: "required" }),
+      geburtsdatumMonat: z.string().min(1, { message: "required" }),
+      geburtsdatumJahr: z.string().min(1, { message: "required" }),
```
[the remark this answers is not in the document]
Comment (chohner) [reply app/services/validation/date.ts:138]: also, I think they probably should be a `z.number().min(1).max(X)`?

File: app/components/formElements/SchemaComponents.tsx [hunk 2]
```diff
@@ -65,6 +66,9 @@ export const SchemaComponents = ({ pageSchema, formComponents }: Props) => (
             />
           );
         }
+        if (nestedSchema.meta()?.description === "date_split") {
```
Comment (aaschlote) [inline app/components/formElements/SchemaComponents.tsx:69]: Out of curiosity, this line is temporary until you figured out this component makes sense to be overall in other pages? Because we have already the component `exclusive_checkbox`, where it was introduced 2 months ago and has been not moved to Strapi.

Asking it because, it sounds the component SchemaComponent has been increased with complexity and I'm wondering if someday this file became some complex as we used to have for the `formular.server.ts`.

File: app/components/formElements/DateSplitInput.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+const DateSplitInput = ({ legend, name }: DateSplitInputProps) => {
+  return (
+    <fieldset className="grid grid-cols-3 gap-4">
+      <legend>{legend}</legend>
+      <Input
+        type="number"
+        placeholder="TT"
+        autoComplete={autocompleteMap[name + ".tag"] ?? "off"}
+        name={name}
+        label="Tag"
```
Comment (Spencer6497) [inline app/components/formElements/DateSplitInput.tsx:18]: should we move these labels to the `translations` file?

File: app/components/formElements/DateSplitInput.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import { autocompleteMap } from "~/util/autocompleteMap";
+import Input, { type InputProps } from "./Input";
+
+export type DateSplitInputProps = InputProps & {
+  legend?: string;
+  name: string;
+};
+
+const DateSplitInput = ({ legend, name }: DateSplitInputProps) => {
```
Comment (Spencer6497) [inline app/components/formElements/DateSplitInput.tsx:9]: nit: could we rename to `SplitDateInput`?

File: app/services/validation/date.ts [hunk 5]
```diff
@@ -52,3 +120,58 @@ export const createDateSchema = (args?: {
       }
     });
 };
+export const createDateSplitSchema = (args?: {
```
Comment (Spencer6497) [inline app/services/validation/date.ts:123]: nit: could we rename to 
```suggestion
export const createSplitDateSchema = (args?: {
```

**Review by Spencer6497 — APPROVED** [review]

File: app/components/formElements/SplitDateInput.tsx [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+        <p className="ds-label-01-reg mb-24">
+          {translations.splitDateComponent.hintText.de}
+        </p>
+
+        <div className="grid grid-cols-4 gap-16 w-full">
+          <div className="col-span-1">
+            <InputLabel id={name + ".tag"}>
+              {translations.splitDateComponent.tagInputLabel.de}
+              <input
+                {...dayField.getInputProps({
```
Comment (aaschlote) [inline app/components/formElements/SplitDateInput.tsx:45]: maybe you can add the `aria-required`. The value can be default to true until we move the component to Strapi

File: app/components/formElements/SplitDateInput.tsx [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+        {/* show error messages for each field with error*/}
+        {hasError && (
+          <div id={errorId}>
+            {dayError && <InputError id={errorId}>{dayError}</InputError>}
+            {monthError && <InputError id={errorId}>{monthError}</InputError>}
+            {yearError && <InputError id={errorId}>{yearError}</InputError>}
+          </div>
+        )}
+        {/* show one error message for one field with error*/}
+        {/* {hasError && (
```
Comment (aaschlote) [inline app/components/formElements/SplitDateInput.tsx:113]: can we remove this comment block?

File: app/services/validation/__test__/date.test.ts [hunk 8]
```diff
@@ -48,3 +51,120 @@ describe("date validation", () => {
...
+    test.each(cases)(
+      "given $input, returns $expected",
+      ({ input, expected }) => {
+        const actual = createSplitDateSchema().safeParse(input);
+        expect(actual).toEqual({ data: expected, success: true });
+      },
+    );
+  });
+
+  describe("failing cases", () => {
```
Comment (aaschlote) [inline app/services/validation/__test__/date.test.ts:81]: maybe we can add a test to negative values

File: app/components/formElements/SplitDateInput.tsx [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+                  "has-error": yearError,
+                })}
+              />
+            </InputLabel>
+          </div>
+        </div>
+        {/* show error messages for each field with error*/}
+        {hasError && (
+          <div id={errorId}>
+            {dayError && <InputError id={errorId}>{dayError}</InputError>}
```
Comment (aaschlote) [inline app/components/formElements/SplitDateInput.tsx:107]: `dayError` will print the error code, shouldn't we translate it?

File: app/components/formElements/SplitDateInput.tsx [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+import InputError from "./InputError";
+import InputLabel from "./InputLabel";
+import { translations } from "~/services/translations/translations";
+import classNames from "classnames";
+
+export type SplitDateInputProps = {
+  name: string;
+};
+
+export const SplitDateInput = ({ name }: SplitDateInputProps) => {
```
Comment (aaschlote) [inline app/components/formElements/SplitDateInput.tsx:12]: I would like to see some tests and storybook for this component.

File: app/components/formElements/SplitDateInput.tsx [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+  const dayField = useField(name + ".tag");
```
Comment (aaschlote) [inline app/components/formElements/SplitDateInput.tsx:13]: wondering if we need to use the values in german, can we not keep the component and schema in english? I guess for the userData I'm ok to be in german.

File: app/components/formElements/Input.tsx [hunk 12]
```diff
@@ -20,6 +20,8 @@ export type InputProps = Readonly<{
   width?: FieldWidth;
   helperText?: string;
   charLimit?: number;
+  autoComplete?: string;
```
Comment (aaschlote) [inline app/components/formElements/Input.tsx:23]: I guess we undo these changes, right?

File: app/services/pdf/getGeburtsdatumDate.ts [hunk 13]
```diff
@@ -0,0 +1,10 @@
+export function getGeburtsdatumDate(dateObj?: {
```
[the remark this answers is not in the document]
Comment (ingriddorioschulze) [reply app/services/pdf/getGeburtsdatumDate.ts:1]: True!

File: app/services/validation/date.ts [hunk 14]
```diff
@@ -52,3 +52,58 @@ export const createDateSchema = (args?: {
...
+            return NaN;
+          }
+          return val;
+        },
+        z.coerce
+          .number({ message: "required" })
+          .refine((val) => !isNaN(val), { message: "invalid_year_format" })
+          .refine((val) => val > 0, { message: "invalid_year_format" })
+          .min(1900, { message: "year_out_of_range" })
+          .max(new Date().getFullYear(), { message: "year_out_of_range" }),
```
[the remark this answers is not in the document]
Comment (ingriddorioschulze) [reply app/services/validation/date.ts]: Yeah, I guess this will need to be adapted. I will wait for design to come up with how they want the error messages.

File: app/services/validation/date.ts [hunk 15]
```diff
@@ -52,3 +52,58 @@ export const createDateSchema = (args?: {
...
+      ),
+    })
+    .superRefine((data, ctx) => {
+      if (!isValidDate(toDateString(data.day, data.month, data.year))) {
+        ctx.addIssue({
+          code: "custom",
+          message: "invalid_date_format",
+          path: ["month"],
```
[the remark this answers is not in the document]
Comment (ingriddorioschulze) [reply app/services/validation/date.ts:97]: True! I am adding **geburtsdatum** as path aiming to target the whole field
